pkg/operator/cassandra/constants/constants.go

```diff
@@ -68,8 +68,8 @@ const (
 	SharedDirName = "/mnt/shared"
 	PluginDirName = SharedDirName + "/" + "plugins"
 
-	DataDirCassandra = "/var/lib/cassandra/data"
-	DataDirScylla    = "/var/lib/scylla/data"
+	DataDirCassandra = "/var/lib/cassandra"
```

Comment:
Is this change backwards compatible?

Reply:
It should be. 
Essentially, the commit log is written in ephemeral storage and is lost when the Pod dies. This is wrong and bottlenecks Cassandra's speed to the ephemeral disk's speed.
With these changes, the commit log is written on the PersistentVolume and persists across reboots.

Reply:
@yanniszark If currently for existing instances the contents of `/var/lib/cassandra/data` where "directly" written to the "root" of the PV:
```
ROOT OF PV
-old-file-1
-old-file-2
```
and you now change it the mount dir to `/var/lib/cassandra`, it would result in this:
```
ROOT OF PV
-old-file-1
-old-file-2
-data/
  -new-file-1
  -new-file-2
```

Meaning that the `old-file-*` are just "sitting" there not being used as new `new-file-*` are generated which are used for the DB(s), which could lead to data loss when Cassandra doesn't replicate fast enough till the next Pod(s) are going down so that the data is still available enough.

Please make sure this is backwards compatible.

Reply:
You're right, it isn't backwards compatible because:

If we remount an disk with existing data on /var/lib/cassandra instead of /var/lib/cassandra/data, cassandra will not recognise the folder structure. So the way to ensure compatibility would be to copy all the data in the PV under a new data folder. 

Old PV filesystem:
- file1
- file2

New PV Filesystem: 
- data/file1
- data/file2

Let's see how we can move on with this.